session.cpp

```diff
@@ -97,10 +104,29 @@ std::string session::convert_buffer()
 }
 
 
-
-std::string session::get_function_from_url(std::string url)
+//get the longest prefix that matches with what's specified in config. If no match is found, an empty string is returned
+std::string session::get_function_from_url(const std::string original_url)
 {
+  std::string url = original_url + "/"; //so that we can handle the part after the last slash
   std::string function = "";
+  int startPos = url.find("/", 0);
+  int upto = url.length();
+  int lastPos;
+  std::string subUrl;
+  while (true){
+    lastPos = url.rfind("/", upto);
+    if (lastPos == 0){
```

Comment:
Yes, good catch! I will close this pull request and submit another pull request.